runtime/gc_glue_java/ArrayletObjectModel.hpp

```diff
@@ -1138,23 +1138,42 @@ class GC_ArrayletObjectModel : public GC_ArrayletObjectModelBase
 		return layout;
 	}
 
+	/**
+	 * Calculates object size in bytes with header and object's hashcode offset
+	 *
+	 * @param[in] forwardedHeader pointer to the MM_ForwardedHeader instance encapsulating the object
+	 * @param[out] hashcodeOffset hashcode offset of object being moved
+	 *
+	 * @return object size in bytes with header
+	 */
+	MMINLINE uintptr_t
+	calculateObjectSizeAndHashcode(MM_ForwardedHeader *forwardedHeader, uintptr_t *hashcodeOffset)
```

Comment:
this might have Preserved in the name (since it graps info from FH)
don't know, maybe overkill